bundle-workflow/src/test_workflow/test_recorder/test_recorder.py

```diff
@@ -0,0 +1,123 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import shutil
+from typing import Dict
+
+import yaml
+
+from test_workflow.test_recorder.test_recorder_builder import \
+    TestRecorderBuilder
+
+
+class Singleton(type):
+    """Metaclass."""
+    _instances: Dict[type, object] = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        else:
+            logging.info("TestRecorder is a singleton class, retrieving existing instance")
+        return cls._instances[cls]
+
+    def get_instance(cls):
+        if cls._instances is not None:
+            return cls._instances[cls]
+
+
+class TestRecorder(object, metaclass=Singleton):
+    ACCEPTABLE_TEST_TYPES = ["integ-test", "bwc-test", "perf-test"]
+
+    def __init__(self, test_run_id, test_type, location):
```

Comment:
I'm not clear with the benefit of making this singleton. Also, this doesn't seem to handle the parallel test run use case. The plan is to run the integ-test suite in parallel at component level. What if we don't keep this singleton and create an instance of it inside IntegTestSuite itself? That should handle parallel test run use case as well.

Reply:
Removed the singleton pattern.